logline_test.go

```diff
@@ -131,88 +133,436 @@ func TestLogLineRecordConformance(t *testing.T) {
-	b.Run("bench", func(b *testing.B) {
-		for n := 0; n < b.N; n++ {
-			ll := LogLine{FieldSeparator: 44}
-			ll.Parse(text, nil)
+func TestLogLineCustomFields(t *testing.T) {
+	t.Run("parse error to many fields", func(t *testing.T) {
+		values := make([]string, 0, LogLineNumFields)
+		for i := 0; i < int(LogLineNumFields)+1; i++ {
+			values = append(values, "value")
+		}
+		want := []byte(strings.Join(values, ","))
+
+		ll := LogLine{FieldSeparator: ','}
+		err := ll.Parse(want, nil)
+		if err != errLogLineTooManyFields {
+			t.Errorf("got: %s want: %s", err, errLogLineTooManyFields)
+		}
+	})
+
+	t.Run("set/get custom fields", func(t *testing.T) {
+		ll := LogLine{FieldSeparator: ','}
+
+		want := []byte("custom1")
+		idx := LogLineNumFields
+		ll.Set(idx, want)
+		got := ll.Get(idx)
+		if !bytes.Equal(got, want) {
+			t.Errorf("got: %s want: %s", got, want)
+
+		}
+
+		want = []byte("customN")
+		idx = LogLineNumFields + NumFieldsBaker - 1
+		ll.Set(idx, want)
+		got = ll.Get(idx)
+		if !bytes.Equal(got, want) {
+			t.Errorf("got: %s want: %s", got, want)
+		}
+
+		// Custom fields should not be serialized.
+		got = ll.ToText(nil)
+		if !bytes.Equal(got, []byte{}) {
+			t.Errorf("got: %s want: ''", got)
 		}
 	})
 }
 
-func BenchmarkToText(b *testing.B) {
-	b.Run("bench", func(b *testing.B) {
-		ll := LogLine{FieldSeparator: 44}
-		for i := 0; i < 100; i++ {
+func TestLogLineGet(t *testing.T) {
+	t.Run("no set value1", func(t *testing.T) {
+		ll := LogLine{FieldSeparator: ','}
+
+		for i := FieldIndex(0); i < LogLineNumFields+NumFieldsBaker; i++ {
+			got := ll.Get(i)
+			if got != nil {
+				t.Errorf("got: %s, want: ''", got)
+			}
+		}
+	})
+
+	t.Run("no set value2", func(t *testing.T) {
+		ll := LogLine{FieldSeparator: ','}
+		ll.Parse([]byte("value,value,value"), nil)
+
+		for i := FieldIndex(3); i < LogLineNumFields+NumFieldsBaker; i++ {
+			got := ll.Get(i)
+			if len(got) != 0 {
+				t.Errorf("got: %s want: ''", got)
+			}
+		}
+	})
+
+	t.Run("no set value3", func(t *testing.T) {
+		ll := LogLine{FieldSeparator: ','}
+		ll.Set(FieldIndex(0), []byte("value"))
+		ll.Set(FieldIndex(1), []byte("value"))
+		ll.Set(FieldIndex(2), []byte("value"))
+
+		for i := FieldIndex(3); i < LogLineNumFields+NumFieldsBaker; i++ {
+			got := ll.Get(i)
+			if len(got) != 0 {
+				t.Errorf("got: %s want: ''", got)
+			}
+		}
+	})
+
+	t.Run("out of range", func(t *testing.T) {
+		defer func() {
+			if r := recover(); r == nil {
+				t.Errorf("got: %s want: panic", r)
```

Comment:
this is always going to be nil, anyway in the error message we should isaying the user we should have panicked
```suggestion
				t.Errorf("Get(%d) should panic", fidx)
```